lib/connection.js

```diff
@@ -208,6 +208,7 @@ var Connection = function (options, container) {
             && options.servername === undefined && options.host !== undefined) {
             this.options.servername = options.host;
         }
+        this.buffer_size = options.buffer_size;
```

Comment:
I think the option name should be a little bit clearer, e.g. session_buffer_size or similar. Otherwise looks good!

Reply:
Thanks for your quick review

Totally agree, that's way more clear :100:

> There are only two hard things in Computer Science: cache invalidation and naming things.

Reply:
Done + added some unit tests

Reply:
> There are only two hard things in Computer Science: cache invalidation and naming things.
and off-by-one errors!